tree-sitter-xml/src/scanner.c

```diff
@@ -0,0 +1,108 @@
+#include <ctype.h>
+#include <tree_sitter/parser.h>
+#include <wctype.h>
+
+enum TokenType {
+    NonXMLName,
+    CharData,
+    ErrRecovery,
+};
+
+void *tree_sitter_xml_external_scanner_create() { return NULL; }
+
+void tree_sitter_xml_external_scanner_destroy(void *payload) {}
+
+void tree_sitter_xml_external_scanner_reset(void *payload) {}
+
+unsigned tree_sitter_xml_external_scanner_serialize(void *payload,
+                                                    char *buffer) {
+    return 0;
+}
+
+void tree_sitter_xml_external_scanner_deserialize(void *payload,
+                                                  const char *buffer,
+                                                  unsigned length) {}
+
+static void advance(TSLexer *lexer) { lexer->advance(lexer, false); }
+
+bool tree_sitter_xml_external_scanner_scan(void *payload, TSLexer *lexer,
+                                           const bool *valid_symbols) {
+    if (valid_symbols[NonXMLName] && !valid_symbols[ErrRecovery]) {
+        bool advanced_once = false;
+        bool found_x_first = false;
+        if (isalpha(lexer->lookahead) || lexer->lookahead == '_') {
+            if (lexer->lookahead == 'x' || lexer->lookahead == 'X') {
+                found_x_first = true;
+                lexer->mark_end(lexer);
+            }
+            advanced_once = true;
+            advance(lexer);
+        }
+
+        if (advanced_once) {
+            while (isalnum(lexer->lookahead) || lexer->lookahead == '_' ||
+                   lexer->lookahead == ':' || lexer->lookahead == '.' ||
+                   lexer->lookahead == L'·' || lexer->lookahead == '-') {
```

Comment:
```suggestion
                   lexer->lookahead == 183 || lexer->lookahead == '-') {
```

Reply:
Is that just stylistic preference? or am I missing something

Reply:
tree-sitter does the same in the parser.

https://github.com/ObserverOfTime/tree-sitter-xml/blob/b74b74fbceebe32a3d0c44f1fadc62ec43038e52/tree-sitter-xml/src/parser.c#L1724